### DISCUSSION TITLE
[Kavita+] Download/Sync "plan to read" list from Myanimelist and add it to "want to read" and "Collections" in Kavita

### DISCUSSION THREAD

**Discussion by Paraxes** [Kareadita/Kavita#3153] [Ideas]
### Idea Description

As described on Discord: 

Paraxos: "Now that you've fixed the Collection pulling/syncing Majora, do you think it would eventually be a possibility to download/sync the "plan to read" list from Myanimelist? Because I have Manga in 2 languages, I am constantly checking which Manga I actually own in a digital format. And even if you just read in English it would be still super beneficial.

So being able to pull the plan to read list from MAL and maybe convert it into a collection, so that you'd have an automatic and easy overview of what you want to read on Kavita, would be a massive QOL-Feature."

You: "Probably and I assume automatically add those to want to read on Kavita or create a collection as well? If you create a feature request I can probably get it in on the v0.8.4 stable as I'm redoing a lot of Kavita+ stuff to make it work better now that I have a feel for it all."

Paraxos: "Right, preferably both ("Want to read" in Kavita and "Collection"), but since I use CDisplayEX as a reader on my tablet, I basically "abuse" collections as my "plan to read" list since CDisplayEX can't access the WTR-list."

### Idea Category

User Experience

### Duration of Using Kavita

1 Year

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
Took a quick look and it seems feasible.

**Comment by impishlucy** [comment]
Let me chime in and ask for the same thing with AniList, not just MAL

**Comment by majora2007** [comment]
I think the best solution would to go the "Smart Collection" route, like I did with MAL Stacks. This will create a collection that keeps in sync with MAL/AL. Then you can go and add to Want to Read if that's better for you (or script it).

**Comment by majora2007** [comment]
This is now implemented in the nightly branch of v0.8.4 and will be delivered in v0.8.5. Any nightly users can test this out (both MAL and AL was implemented) and it will sync into your Want To Read on Kavita.

https://github.com/Kareadita/Kavita/pull/3507
